components/MultiPanel/index.ts

```diff
@@ -1,12 +1,183 @@
 import './styles.css';
 
+/**
+ * A multi-panel view that the user can add any number of 'panels'. 
+ * 'a panel' consists of two elements. even index element begomes heading, 
+ * and odd index element becomes the expandable content.
+ */
 export default class MultiPanel extends HTMLElement {
+
   constructor () {
     super();
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
   }
 
   connectedCallback () {
+    this._childrenChange();
+
+    const children : Element[] = Array.from(this.children);
+    for (let i : number = 0; i < children.length; i++){
+      const child : Element = children[i]
+
+      if(i % 2 === 0){  
+        // even index = heading element
+        child.classList.add('panel-heading');
+        
+        // for A11y
+        child.id = `panel-heading-${i}`;
```

Comment:
If the child already has an ID we shouldn't overwrite it.

Reply:
Ohh, and you're going to get duplication of IDs if you have two `<multi-panel>`s. Maybe generate a random ID?